### ISSUE TITLE
NullPointerException from probably-should-be-illegal "specialized aspect"

### ISSUE THREAD

**Issue by remexre** [melt-umn/silver#506]
```silver
nonterminal Foo<a>;
synthesized attribute pp::String occurs on Foo<a>, Bar;

production foo
top::Foo<a> ::= x::a
{ top.pp = "asdf"; }

nonterminal Bar;

inherited attribute baz::Integer occurs on Bar;

-- !!!
aspect production foo
top::Foo<Bar> ::= x::Bar
{ x.baz = 0; }

production bar
top::Bar ::=
{ top.pp = toString(top.baz); }

function main
IOVal<Integer> ::= args::[String]  ioIn::IO
{
  return evalIO(do {
    printM(foo(bar()).pp);
    printM(foo("hmmmm").pp);
    return 0;
  }, ioIn);
}
```

Aspects like this should be illegal unless someone can find a way to support them that doesn't require full RTTI.

**Comment by ericvanwyk** [comment]
Is this bug fixed in 4caca02 ?  There is no mention of it there.

**Comment by remexre** [comment]
Yeah, the cause is the weird aspect production, which is now illegal.
